src/clj_gatling/simulation.clj

```diff
@@ -124,66 +134,144 @@
    user-id]
   (let [c (async/chan)
         should-run-now? (if concurrency-distribution
-                          #(let [progress (runners/calculate-progress runner @sent-requests simulation-start)
-                                 target-concurrency (* concurrency
-                                                       (concurrency-distribution progress context))]
-                             (> target-concurrency @concurrent-scenarios))
+                          (let [modifier-fn (if (= 2 (arg-count concurrency-distribution))
```

Comment:
I think it might be better to switch to a map. Adding more parameters makes it harder to expand in the future. I think it was a mistake to use two arity function in the first place. In progress tracker I decided to use map from the beginning. What about something like this:

```
  (let [modifier-fn (if (= 2 (arg-count concurrency-distribution))
                                              (fn [{:keys [progress context]}]
                                                (concurrency-distribution progress context))
                                              concurrency-distribution)]
                            #(let [[progress duration] (runners/calculate-progress runner
                                                                                   @sent-requests
                                                                                   simulation-start
                                                                                   (LocalDateTime/now))
                                   target-concurrency (* concurrency (modifier-fn {:progress progress
                                                                                   :duration duration
                                                                                   :context context}))]
                               (> target-concurrency @concurrent-scenarios)))
```

Reply:
That makes sense. I was trying to stick to the original APi as much as possible so that people wouldn't have to significantly change the way they were writing things, but as long as it's backwards compatible this is a nice pattern. Let me have a look.

Reply:
Updated, including the docs (See README.md, under Options > Ramp-up).